### PR TITLE
Move log sanitization to implementable

### PR DESCRIPTION
**Pull request by serathius**
/cc @44past4 @dashpole @logicalhan @wojtek-t @ehashman @brancz

### CODE REVIEW

**Comment by dashpole** [comment]
/approve
/hold
After @wojtek-t finishes the PRR, feel free to remove the hold.

Did you get any input from sig-testing on the test plans?  If not, it would be good to have someone ack them.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:209]: What about any potential performance degradations?

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:335]: I disagree that this is not applicable.

One pitfall I could anticipate is if there is a bug in log sanitization that causes log entries to be mangled and hence compromises the integrity of log data. I'd say it's a low risk that's addressed by having the command line flag to disable it, but I think the risk is there.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:337]: Loss of log integrity.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:342]: Assuming e2e tests pass with both the CLI flag enabled/disabled, I think this is covered.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:346]: Yes, please mention the CLI flags here.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:355]: Command line arguments of running control plane processes.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:364]: N/A

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:373]: N/A

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:378]: N/A, this is a logging change.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:382]: Doesn't need to be completed yet as this is in alpha.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:450]: If log write time is increased then I'd say this does potentially affect existing SLIs/SLOs, but the likelihood of significant performance degradation is low.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:461]: :+1:

File: keps/sig-instrumentation/1753-logs-sanitization/kep.yaml
Comment (ehashman) [inline keps/sig-instrumentation/1753-logs-sanitization/kep.yaml:12]: You can add me as well if you like as either a reviewer or approver.

**Review by ehashman — CHANGES_REQUESTED** [review]
Some minor comments, caught some small bits that are incomplete. Will review again on Monday.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (serathius) [reply keps/sig-instrumentation/1753-logs-sanitization/README.md:209]: I don't think we should add specific mechanism to monitor performance degradation and can rely on existing qualification done by scalability team. 

Before enabling this feature by default we will work with SIG Scalability on qualifying (e.g. before major changes for Structured Logging we requested to run scalability tests on PR before merging)

Before feature is enabled by default, performance should not be big concern we should just ensure that e2e periodic test doesn't fail due to overhead

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (serathius) [reply keps/sig-instrumentation/1753-logs-sanitization/README.md:335]: Ok, I see your point. Maybe let's revisit this during graduation to beta?

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (serathius) [reply keps/sig-instrumentation/1753-logs-sanitization/README.md:450]: As section below focuses on increased resources there.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (wojtek-t) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:295]: Enabling log sanitization (not swtiching between log formats) :)

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (wojtek-t) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:317]: This one is always applicable :)

The feature will simply start working again :)

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (wojtek-t) [inline keps/sig-instrumentation/1753-logs-sanitization/README.md:325]: Can you add: "We will run manual tests before enabling the feature by default".

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (wojtek-t) [reply keps/sig-instrumentation/1753-logs-sanitization/README.md:390]: It potentially can for the logs that are synchronous (due to what Elana mentioned) [so error ones, right?]

I agree it's low risk, but I think it's worth mentioning.

**Review by wojtek-t — COMMENTED** [review]
Couple smaller comments.

File: keps/sig-instrumentation/1753-logs-sanitization/README.md
Comment (wojtek-t) [reply keps/sig-instrumentation/1753-logs-sanitization/README.md:209]: It's not a matter of just "not failing" - we shouldn't eat all the slack we have with that. It's a matter of overhead.

But I agree that I wouldn't add a dedicated stuff for it per-se.

**Comment by serathius** [comment]
@dashpole, I reached out to sig-testing on #slack on Friday, but didn't get any response. Will try again

**Comment by serathius** [comment]
/cc @BenTheElder @spiffxp @stevekuznetsov 
PTAL

**Comment by serathius** [comment]
ping @ehashman

**Review by ehashman — APPROVED** [review]
/lgtm

**Comment by ehashman** [comment]
Waiting on SIG Testing to remove hold, this is GTG from the Instrumentation side.

**Review by spiffxp — APPROVED** [review]
/lgtm
as SIG Testing chair

I'm going to assume extensive unit and integration testing comes first, so we're not just relying on e2e tests. I otherwise agree with what I see here as far as test plan and evaluating potential performance impact

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/enhancements/pull/2052#pullrequestreview-502399740 +1

**Comment by lavalamp** [comment]
Did anyone from API Machinery review this? I really think we should make sure this mechanism doesn't collide with things we want to do in the future, and it looks like it might. @deads2k @liggitt

**Comment by lavalamp** [comment]
Also SIG Auth, and someone from the API review/approval process from SIG Arch?

**Comment by lavalamp** [comment]
@apelisse / @sttts can you review this?

**Pull request merged**
